AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -333,6 +333,23 @@ - (void)fetchData
   }
 }
 
+#pragma mark - Layout and Sizing
+
+- (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
+{
+  ASDN::MutexLocker l(__instanceLock__);
+
+  // If the image node is currently in the loading process and no valid size is applied return CGSizeZero for the time
+  // being.
+  // TODO: After 2.0 is stable we should remove this behavior as a ASNetworkImageNode is a replaced element and the
+  // client code should set the size of an image or it's container it's embedded in
+  if (ASIsCGSizeValidForSize(constrainedSize) == NO && _URL != nil && self.image == nil) {
```

Comment:
@garrettmoon @appleguy Put up a PR that add's an assertion if the network image node calculates it's size and `_URL == nil` with invalid constrained size or no image set: #2389